src/data/equiv/transfer_instance.lean

```diff
@@ -121,103 +121,88 @@ end
 /-- Transfer `semigroup` across an `equiv` -/
 @[to_additive "Transfer `add_semigroup` across an `equiv`"]
 protected def semigroup [semigroup β] : semigroup α :=
-{ mul_assoc := by simp [mul_def, mul_assoc],
-  ..equiv.has_mul e }
+let mul := e.has_mul in
+by resetI; apply e.injective.semigroup _; intros; exact e.apply_symm_apply _
```

Comment:
Did you `#print` the resulting definitions? Sometimes tactic proofs insert unwanted `id`s and this makes it hard for Lean to match two instances.

Reply:
`#print equiv.comm_semigroup` gives:
```
  let mul : has_mul α := e.has_mul
  in function.injective.comm_semigroup ⇑e _ _
```
Do you think the `let` is problematic?

Reply:
I don't know. Ask @gebner .

Reply:
Another option might be to use `transport`, which I think should work throughout this file, and hopefully(!) gives good definitions.

Reply:
Separately, let's reduce the unneccesary use of `;` here. :-)

Reply:
The let is fine here.